OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/ResetPasswordAlertDialogFragment.java

```diff
@@ -0,0 +1,190 @@
+         * and letting submit() control the logic so, we are overriding View.OnClickListener here.
+         */
+        ((AlertDialog) getDialog()).getButton(DialogInterface.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                submit(editText.getText().toString().trim());
+            }
+        });
+    }
+
+    public void setListener(@NonNull OnActivityResultListener listener) {
+        this.listener = listener;
+    }
+
+    public void showError(@NonNull String error) {
+        textInputLayout.setError(error);
+        textInputLayout.setErrorEnabled(true);
+        textInputLayout.sendAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_FOCUSED);
+    }
+
+    private void submit(@Nullable String email) {
+        if (!NetworkUtil.isConnected(getContext())) {
+            showError(getString(R.string.network_not_connected_short));
+        } else if (!InputValidationUtil.isValidEmail(email)) {
+            showError(getString(R.string.error_invalid_email));
+        } else {
+            loginService.resetPassword(email).enqueue(new ErrorHandlingCallback<ResetPasswordResponse>(
+                    getContext(), CallTrigger.USER_ACTION) {
+                @Override
+                protected void onResponse(@NonNull final ResetPasswordResponse result) {
+                    if (listener != null) {
+                        if (result.isSuccess()) {
+                            listener.onActivityResult(0, Activity.RESULT_OK, null);
+                            dismiss();
+                        } else {
+                            final String errorMsg = result.getPrimaryReason();
+                            final Intent data = new Intent();
+                            data.putExtra("error", errorMsg);
+                            listener.onActivityResult(0, Activity.RESULT_CANCELED, data);
+                            showError(errorMsg);
+                        }
+                    }
+                }
+
+                @Override
+                protected void onFailure(@NonNull Throwable error) {
+                    if (listener != null) {
+                        final String errorMsg = ErrorUtils.getErrorMessage(error, getContext());
+                        final Intent data = new Intent();
+                        data.putExtra("error", errorMsg);
+                        listener.onActivityResult(0, Activity.RESULT_CANCELED, data);
+                        showError(errorMsg);
+                    }
+                }
+            });
+        }
+    }
+
+    @Nullable
+    @Override
+    protected ButtonAttributes getNegativeButtonAttributes() {
+        return new ButtonAttributes() {
+            @NonNull
+            @Override
+            public String getMessage() {
+                return getContext().getResources().getString(R.string.label_cancel);
```

Comment:
I copied and pasted a few of my unaddressed comments from Marc's previous attempted implementation at #817 😜